arch/posix/core/cpuhalt.c

```diff
@@ -20,8 +20,7 @@
  * A weak stub for sys_arch_reboot(), which does nothing
  */
 
-#include "posix_core.h"
-#include "posix_soc_if.h"
```

Comment:
Nit: I'm from the school that prefers that if a header is needed by a file, that file includes the header directly instead of expecting that header be included indirectly by some other header.
I would be happier if `posix_soc_if.h` was included directly (with whatever path it needs), but I will not be blocking this PR because of such a thing.